examples/hackernews/front-end/src/App.tsx

```diff
@@ -66,41 +66,47 @@ function Feed() {
 
   // Reactive version:
   const skipclient: Client = useContext(SkipContext)!;
-  useEffect(() => {
-    async function getInitialData() {
-      // TODO: Simplify this.
-      const pubkey = String.fromCharCode.apply(null, [
-        ...new Uint8Array(await Protocol.exportKey(skipclient.creds.publicKey)),
-      ]);
-      const response = await fetch(BASE_URL, {
-        headers: {
-          "X-Reactive-Auth": btoa(pubkey),
-        },
-      });
-      const data = await response.json();
-      setPosts(data);
-      const reactiveToken = JSON.parse(
-        response.headers.get("Skip-Reactive-Response-Token")!,
-      );
-      // TODO: Make this happen transparently in the skip client.
-      const reactiveCollection = reactiveToken.collection;
-      const reactiveWatermark = BigInt(reactiveToken.watermark);
-      skipclient.subscribe(
-        reactiveCollection,
-        reactiveWatermark,
-        (updates, _isInit) => {
-          const updatedPosts = updates[0][1]! as Post[];
-          setPosts(updatedPosts);
-        },
-      );
-    }
-
-    try {
-      getInitialData();
-    } catch (error) {
-      console.error(error);
-    }
-  }, []);
+  useEffect(
+    () => {
+      async function getInitialData() {
+        // TODO: Simplify this.
+        const pubkey = String.fromCharCode.apply(null, [
+          ...new Uint8Array(
+            await Protocol.exportKey(skipclient.creds.publicKey),
+          ),
+        ]);
+        const response = await fetch(BASE_URL, {
+          headers: { "X-Reactive-Auth": btoa(pubkey) },
+        });
+        const data = await response.json();
+        setPosts(data);
+        const reactiveToken = JSON.parse(
+          response.headers.get("Skip-Reactive-Response-Token")!,
+        );
+        // TODO: Make this happen transparently in the skip client.
+        const reactiveCollection = reactiveToken.collection;
+        const reactiveWatermark = BigInt(reactiveToken.watermark);
+        skipclient.subscribe(
+          reactiveCollection,
+          reactiveWatermark,
+          (updates, _isInit) => {
+            const updatedPosts = updates[0][1]! as Post[];
+            setPosts(updatedPosts);
+          },
+        );
+      }
+
+      try {
+        getInitialData();
+      } catch (error) {
+        console.error(error);
+      }
+    },
+    // Specify no dependencies to react since the callback is meant to only
+    // execute once to fetch the initial data.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [],
```

Comment:
Here, the linter wants
```suggestion
    [skipclient],
```

Reply:
In this case it is not clear to me that it is ok to execute the body of the `useEffect` more than once. Here the proposed dep `skipclient` is also constant, so maybe it doesn't matter. I am unsure, are others confident?

Reply:
Same as for the other comment thread -- I think the linter is right that `skipclient` is not necessarily constant in general, as it comes from React's `useContext`.   And if that Skip Client context changes, then it would be right to reevaluate this effect, so I think the linter-proposed dependency array looks right.  But confirmation from Daniel would be good :)